Improve error messages for bad server behaviour

HugoArts opened the pull request:
This implements nice and helpful error messages if the server has a 500 error, returns invalid JSON, or does other strange things that generally shouldn't happen.

I have refactored the `_get`, `_put`, `_post` methods into a single `_request` method, updated all tests, and added a few new tests to keep our coverage up.

This fixes issues #3 and #4

HugoWend committed:
improve error messages for server snafus

The main goal of this patch is to improve error messages when the server
encounters a 500 or doesn't return any JSON data in an expected format.

To do this I also refactored the _get, _put, _post methods into a single
_request method that takes a `method` argument, since the code for them
was almost exactly identical. This did require me to change a whole
bunch of tests, unfortunately.

HugoWend committed:
added more tests to get coverage back up to 100%

HugoWend committed:
fix forgotten format call

HugoWend committed:
don't assume 200 is the only non-error status code

dporru commented on pluvo/pluvo.py:
```diff
@@ -140,52 +139,44 @@ def _set_auth_params(self, params=None):
+        except ValueError:
+            if r.status_code == 500:
+                msg = ('Server returned a 500 error. This is likely a bug. '
+                       'Contact us at https://github.com/wendbv/pluvo/')
```
This is not a public repo. Probably better to use https://github.com/wendbv/pluvo-python

dporru commented on pluvo/pluvo.py:
```diff
@@ -140,52 +139,44 @@ def _set_auth_params(self, params=None):
+                msg = ('Server returned a non 20x status code, but the '
+                       'returned JSON contains no error message. This is '
+                       'likely a bug. Contact us at '
+                       'https://github.com/wendbv/pluvo/\n\nreturned JSON: '
```
Same thing about public repo

HugoWend committed:
don't tell users to submit bug reports to a private repo

HugoArts commented:
@dporru this ok to merge now?

HugoWend committed:
add response body to PluvoAPIException class

Especially now that more error information is being added to Pluvo
responses, this should be useful.

dporru commented:
:shipit:

HugoArts commented:
too late I'm adding more stuff now 😅
